File: pw6/Agence.java
import java.util.ArrayList;
import java.util.List;

public class Agence {
    private String name;
    private List<Vehicule> fleet;

    public Agence(String name) {
        this.name = name;
        this.fleet = new ArrayList<>();
    }

    public void ajouterFlotte(Vehicule v) {
        fleet.add(v);
    }

    public void afficherFlotte() {
        for (Vehicule v : fleet) {
            v.afficher();
        }
    }

    public void close() {
        System.out.println(" -> The agency " + name + " closes its doors.");
    }
}

File: pw6/Agence.cpp
#include "Agence.h"
#include <iostream>

Agence::Agence(std::string n) : name(n) {}

Agence::~Agence() {
    std::cout << " -> The agency " << name << " closes its doors." << std::endl;
    for (Vehicule* v : fleet) {
        delete v;
    }
}

void Agence::ajouter_flotte(Vehicule* v) {
    fleet.push_back(v);
}

void Agence::afficher_flotte() const {
    for (Vehicule* v : fleet) {
        v->afficher();
    }
}
